src/channeldef.ts

```diff
@@ -198,6 +199,72 @@ export type TextDef<F extends Field> =
   | FieldOrDatumDefWithCondition<StringDatumDef<F>, Text>
   | ValueDefWithCondition<StringFieldDef<F>, Text>;
 
+/**
+ * Position of a label relative to its base mark's position, which includes anchor direction (anchor) and distance (offset).
+ */
+export type LabelPosition = {
+  /**
+   * distance of a label from its base mark.
+   * if negative, the label position is insize.
+   */
+  offset: number;
+
+  /**
+   * anchor direction of a label from its base mark.
+   * The value could be one of `'left '` , `'right '` , `'top '` , `'middle '` ,
```

Comment:
Please use double quotation marks ("), not single (') for all docs.

Reply:
Also, don't put spaces before the last ".